.github/workflows/ci.yml

```diff
@@ -20,29 +20,34 @@ jobs:
       fail-fast: false
       matrix:
         os: [ubuntu-latest, macos-latest, windows-latest]
-        node: [14, 16, 18]
+        node: [16, 18, 20]
     runs-on: ${{ matrix.os }}
     steps:
-    - uses: actions/checkout@v3
+    - uses: actions/checkout@v4
+    # https://github.com/nodejs/node-gyp#installation
+    - name: Use Python 3.11
+      uses: actions/setup-python@v4
+      with:
+        python-version: '3.11'
     - name: Use Node.js ${{ matrix.node }}
       uses: actions/setup-node@v3
       with:
         cache: yarn
         node-version: ${{ matrix.node }}
         check-latest: true
     - name: Install Dependencies
-      run: yarn install
+      run: yarn global add node-gyp && yarn install --frozen-lockfile
```

Comment:
Why does node-gyp need to be a global? 

Can we add it as a `devDependency` instead?

Reply:
It's already listed there, but somehow it seems that Windows sometimes needs it in a global environment. I haven't investigated whether this is a problem with the package manager resolver or some kind of implementation, but generally there shouldn't be a major problem